vcd/resource_vcd_lb_app_rule_test.go

```diff
@@ -152,6 +152,7 @@ data "vcd_lb_app_rule" "test" {
   vdc          = "{{.Vdc}}"
   edge_gateway = "{{.EdgeGateway}}"
   name         = vcd_lb_app_rule.test.name
+  depends_on   = [vcd_lb_app_rule.test]
```

Comment:
What is the reason for adding all these `depends_on` which were previously not there?

Reply:
All the tests where we have a resource and a data source for the same entity are now failing. We have seen this behavior when we ran the binary tests in 3.0 release. 
Since the tests now run with the `terraform` binary, also the inline tests with the same construct are failing.
We can either re-write these tests to use separate steps, but that would be quite a lot of code, with no immediate benefit.
Since this paradigm (resource + data source in the same configuration) only makes sense in our tests (there is no need for it in a production environment), this is the quickest fix

Reply:
It would be helpful to have a comment somewhere in the code explaining this, as it's not very intuitive.